web3/exceptions.py

```diff
@@ -30,6 +30,16 @@ class Web3Exception(Exception):
             # deal with other exceptions
     """
 
+    def __init__(
+        self,
+        *args: Any,
+        user_message: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        super().__init__(*args)
+        # Assign properties of Web3Exception
+        self.__dict__.update(locals())
```

Comment:
Sorry I missed how we were updating this the first time around. Is there a reason we are using `locals()` here instead of just assigning `self.user_message = user_message`?

Reply:
I did not use user_message explicitly, though I could. `locals()` here also saves the other args and kwargs on the `Web3Exception`. `kwargs` are not supported by `Exception` as far as I could tell so this felt like a good place to keep that data.

Also thought keeping it open to anything the user would want to store might be useful. If we want to enforce it strictly to `user_message` I am fine doing so. Not sure if there are risks to saving any named property on the `Web3Exception`.